nilearn/datasets/func.py

```diff
@@ -1279,3 +1282,189 @@ def fetch_mixed_gambles(n_subjects=1, data_dir=None, url=None, resume=True,
+    who are accessing shared copies of this data. If you are sharing HCP Open
+    Access data, please advise your co-researchers that they must register with
+    ConnectomeDB and agree to these terms.
+    Register and sign the Open Access Data Use Terms at
+    ConnectomeDB: https://db.humanconnectome.org/
+    """
+    url = "http://www.nitrc.org/frs/download.php/8037/Megatrawls.tgz"
+    opts = {'uncompress': True}
+
+    # standard dataset terms
+    dimensionalities = [25, 50, 100, 200, 300]
+    timeseries_methods = ['multiple_spatial_regression', 'eigen_regression']
+    output_matrices_names = ['correlation', 'partial_correlation']
+
+    if dimensionality is not None:
+        dimensionality = check_parameters_megatrawls_datasets(
+            dimensionality, dimensionalities, 'dimensionality')
+    else:
+        dimensionality = dimensionalities
+
+    if timeseries is not None:
+        timeseries = check_parameters_megatrawls_datasets(
+            timeseries, timeseries_methods, 'timeseries')
+    else:
+        timeseries = timeseries_methods
+
+    if matrices is not None:
+        matrices = check_parameters_megatrawls_datasets(
+            matrices, output_matrices_names, 'matrices')
+    else:
+        matrices = output_matrices_names
+
+    dataset_name = 'Megatrawls'
+    data_dir = _get_dataset_dir(dataset_name, data_dir=data_dir, verbose=verbose)
+    description = _get_dataset_descr(dataset_name)
+
+    # Generate all combinations
+    dims, tseries, mats = list(zip(*list(itertools.product(dimensionality, timeseries, matrices))))
+    files = []
+    ids_correlation = []
+    ids_partial = []
+    timeseries_map = dict(multiple_spatial_regression='ts2', eigen_regression='ts3')
+    matrices_map = dict(correlation='Znet1.txt', partial_correlation='Znet2.txt')
+    for index, (dim, tserie, mat) in enumerate(zip(dims, tseries, mats)):
+        if mat == 'correlation':
+            ids_correlation.append(index)
+        elif mat == 'partial_correlation':
+            ids_partial.append(index)
+        filepath = os.path.join(
+            '3T_Q1-Q6related468_MSMsulc_d%d_%s' % (dim, timeseries_map[tserie]), matrices_map[mat])
+        files.append((filepath, url, opts))
+
+    # Fetch all the files
+    files = _fetch_files(data_dir, files, resume=resume, verbose=verbose)
+
+    # Load the files into arrays
+    correlation = [csv_to_array(files[id_c]) for id_c in ids_correlation]
+    partial = [csv_to_array(files[id_p]) for id_p in ids_partial]
+    # Taking the account of all the given dimensions & timeseries
+    # methods to give the end users to identify themselves about the
+    # matrices which are fetched.
+    dimensions_correlation = [dims[id_c] for id_c in ids_correlation]
+    dimensions_partial = [dims[id_p] for id_p in ids_partial]
+    timeseries_correlation = [tseries[id_c] for id_c in ids_correlation]
+    timeseries_partial = [tseries[id_p] for id_p in ids_partial]
```

Comment:
I find the output overly complicated. Why splitting everything between `correlation` and `partial_correlation` and not having three lists in the bunch: `dimensionality`, `timeseries` and `matrices`? I am still convinced that allowing only one value per parameter would be much simpler: this is what we already do for most of the fetchers (for harvard oxford, only one type of atlas is allowed, for aal, it's the same, only one version at a time).